libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Generic.cpp

```diff
@@ -147,31 +147,66 @@ bool AP_BattMonitor_SMBus_Generic::check_pec_support()
         return false;
     }
 
-    // extract version
-    uint8_t version = (data & 0xF0) >> 4;
-
-    // version less than 0011b (i.e. 3) do not support PEC
-    if (version < 3) {
-        _pec_supported = false;
-        _pec_confirmed = true;
-        return true;
-    }
-
     // check manufacturer name
     uint8_t buff[AP_BATTMONITOR_SMBUS_READ_BLOCK_MAXIMUM_TRANSFER + 1] {};
     if (read_block(BATTMONITOR_SMBUS_MANUFACTURE_NAME, buff, sizeof(buff))) {
         // Hitachi maxell batteries do not support PEC
         if (strcmp((char*)buff, "Hitachi maxell") == 0) {
             _pec_supported = false;
-            _pec_confirmed = true;
+            _v_multiplier = 1;
+            _i_multiplier = 1;
+            _specification_info_confirmed = true;
             return true;
         }
     }
+    // extract version
+    uint8_t version = (data & 0xF0) >> 4;
+
+    // version less than 0011b (i.e. 3) do not support PEC
+    if (version < 3) {
```

Comment:
Hi, thanks for this. I guess we could shorten this to a single line, "_pec_supported = version >= 3".  Alternatively just move the "else {" up one line to make it conform with AP's regular formatting style.